nextcord/http.py

```diff
@@ -204,33 +653,56 @@ async def ws_connect(self, url: str, *, compress: int = 0) -> Any:
 
         return await self.__session.ws_connect(url, **kwargs)
 
+    # ruff: noqa: PLR0915
+    # Ruff says this is too many statements in a single function, but I'm not sure how to break it down while
+    #  maintaining readability. I suppose general purpose rules cannot be perfectly applied in every situation.
     async def request(
         self,
         route: Route,
         *,
         files: Optional[Sequence[File]] = None,
         form: Optional[Iterable[Dict[str, Any]]] = None,
+        auth: Optional[str] = MISSING,
+        retry_request: bool = True,
         **kwargs: Any,
     ) -> Any:
-        bucket = route.bucket
-        method = route.method
-        url = route.url
-
-        lock = self._locks.get(bucket)
-        if lock is None:
-            lock = asyncio.Lock()
-            self._locks[bucket] = lock
-
-        # header creation
-        # user agent is provided by our aiohttp client already
-        headers: Dict[str, str] = {}
-
-        if self.token is not None:
-            headers["Authorization"] = "Bot " + self.token
-        # some checking if it's a JSON request
-        if "json" in kwargs:
-            headers["Content-Type"] = "application/json"
-            kwargs["data"] = utils.to_json(kwargs.pop("json"))
+        """|coro|
+
+        Makes an API request to Discord, handling authorization (if needed), rate limits, and limited error handling.
+
+        Parameters
+        ----------
+        route: :class:`Route`
+            The Discord Route to make the API request for.
+        files: Optional[Sequence[:class:`File`]]
+            The files to provide with the JSON body of this request. If provided, the body of the request will be
+            repackaged into a multipart form.
+        form: Optional[Iterable[:class:`dict`[:class:`str`, `Any`]]]
+            pass
+        auth: :class:`str` | `None` | `MISSING`
```

Comment:
```suggestion
        auth: :class:`str` | `None`
```
`MISSING` shouldn't be included in the type hint since it's not really a type (for now).

Reply:
Also this shouldn't be a 3.10 union.